### PR TITLE
Changes 25: New Changes Class

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Description

**This is replicating https://github.com/getkirby/kirby/pull/6681**

### Summary of changes

- New `Kirby\Content\Changes` class to track changed models in the `site.txt`
- The Changes class is used in the Changes controller to track and untrack changed versions.

### Enhancements

- New `Kirby\Content\Changes` class to track changed models in the `site.txt`

## Ready?

- [x] In-code documentation (wherever needed)
- [x] Unit tests for fixed bug/feature
- [ ] Tests and CI checks all pass

### For review team

- [x] Add changes & docs to release notes draft in Notion

### CODE REVIEW

**Comment by distantnative** [comment]
Another thought: we could also stores this in a dedicated file in the `site/log/` root instead of the site content file.

**Comment by bastianallgeier** [comment]
I thought about this as well. Storing it in the site.txt really isn't that great after all. I was also thinking about a new /site/accounts/users.txt 

I like site/log, but my only concern is that it would be a new directory that needs write permissions.

**Comment by distantnative** [comment]
I do like `/site/accounts/users.txt` - could be an issue if the directory is excluded from uploads/git... etc. though?

**Comment by bastianallgeier** [comment]
That's basically why I still went for the site.txt after all. Each alternative solution has potential pitfalls that we need to think through. That's why I thought I really want to get the functionality done first and we can still change the location later. 

I know that there are a lot of "let's think about this later" points at the moment, but I honestly feel like we won't ever get this done if we don't approach it more pragmatically right now.

**Comment by distantnative** [comment]
Ok, let's leave it with the `site.txt`.

File: src/Content/Changes.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	 */
+	public function untrack(ModelWithContent $model): void
+	{
+		$changes = $this->field()->yaml();
+		$uuid    = (string)$model->uuid();
+		$index   = array_search($uuid, $changes);
+
+		if ($index !== false) {
+			unset($changes[$index]);
+		}
```
Comment (distantnative) [inline src/Content/Changes.php]: What do you think https://github.com/getkirby/kirby/pull/6681#discussion_r1766708369 ?

Comment (bastianallgeier) [reply src/Content/Changes.php]: ✅

**Comment by bastianallgeier** [comment]
The test coverage report is again completely wrong. I have no idea why this is failing so often to even recognize the @covers annotations. So annoying.

**Review by distantnative — APPROVED** [review]
Me neither. There are factual unit tests covering the methods marked as uncovered.

**Pull request merged**